src/runner.js

```diff
@@ -28,12 +28,96 @@ const debugResponse = _debug('remote-s3:Runner:res');
  * actually transmitted matched a pre-computed size and hash
  */
 class Runner {
```

Comment:
Because the `Client` is the thing that runs all client requests, it prepares uploads and executes downloads.  The `Controller` is the thing that does all the S3 requests.  They both need to run `{url, method, headers}` style request, so they share the ability to run them through the `Runner` class.